### PR TITLE
Add link to v2 migration document in readme for main branch

### PR DESCRIPTION
**Pull request by brycewang-microsoft**

### CODE REVIEW

**Review by timtay-microsoft — APPROVED** [review]

File: readme.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Microsoft Azure IoT SDK for .NET
 
+## 2.0.0 clients release notice
+
+A [2.0.0 preview release](https://github.com/Azure/azure-iot-sdk-csharp/releases/tag/2022-10-14-v2preview) has been published for each package in this library. 
+With this release, all future features will be brought only to this new major version, so users are highly encouraged to migrate from the 1.X.X releases to 
+the new 2.0.0 releases.
```
Comment (abhipsaMisra) [inline readme.md:7]: We might want to word this a bit differently, given that the v2 is currently in preview. 
How about we say something like - A 2.0.0 preview prelease has been published for each package in this library. These packages simplify usage and handling of our clients, and we highly encourage you to try out these packages and share your feedback. Once we promote these changes to `main` in the future, all future features will be brought only to this new major version.

Thoughts?

File: readme.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
 # Microsoft Azure IoT SDK for .NET
 
+## 2.0.0 clients release notice
```
Comment (drwill-ms) [inline readme.md:3]: I don't think it is necessary to specify the minor and patch version. Just v2 should be sufficient.

File: readme.md [hunk 3]
```diff
@@ -1,5 +1,13 @@
...
+
+A [2.0.0 preview release](https://github.com/Azure/azure-iot-sdk-csharp/releases/tag/2022-10-14-v2preview) has been published for each package in this library.
```
Comment (drwill-ms) [inline readme.md:5]: If you do want to specify this exact release, then it should be 2.0.0-preview001. And then I'd say keep an eye out for updates.

File: readme.md [hunk 4]
```diff
@@ -1,5 +1,13 @@
 # Microsoft Azure IoT SDK for .NET
 
+## 2.0.0 clients release notice
+
+A [2.0.0 preview release](https://github.com/Azure/azure-iot-sdk-csharp/releases/tag/2022-10-14-v2preview) has been published for each package in this library. 
+With this release, all future features will be brought only to this new major version, so users are highly encouraged to migrate from the 1.X.X releases to
```
Comment (drwill-ms) [inline readme.md:6]: Don't encourage users to migrate while it is in preview.

Instead, we want them to try it out and give us feedback.

**Review by drwill-ms — CHANGES_REQUESTED** [review]
Comments inline

File: readme.md (hunk 1 above)
Comment (brycewang-microsoft) [reply readme.md:7]: Yeah, I like it!

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**